x-pack/plugins/lens/public/xy_visualization_plugin/xy_config_panel.test.tsx

```diff
@@ -0,0 +1,365 @@
+    expect(testSetTitle('There!')).toMatchObject({
+      x: { title: 'There!' },
+    });
+  });
+
+  test('the x dimension panel accepts any operations', () => {
+    const datasource = {
+      ...mockDatasource(),
+      renderDimensionPanel: jest.fn(),
+    };
+    const state = testState();
+    const component = mount(
+      <XYConfigPanel
+        datasource={datasource}
+        setState={jest.fn()}
+        state={{ ...state, x: { ...state.x, accessor: 'shazm' } }}
+      />
+    );
+
+    const panel = testSubj(component, 'lnsXY_xDimensionPanel');
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const nativeProps = (panel as any).nativeProps;
+    const { columnId, filterOperations } = nativeProps;
+    const ops = [
+      { dataType: 'number' },
+      { dataType: 'string' },
+      { dataType: 'boolean' },
+      { dataType: 'date' },
+    ];
+    expect(columnId).toEqual('shazm');
+    expect(ops.filter(filterOperations)).toEqual(ops);
+  });
+
+  test('allows toggling the x axis gridlines', () => {
+    const toggleXGridlines = (showGridlines: boolean) => {
+      const setState = jest.fn();
+      const state = testState();
+      const component = mount(
+        <XYConfigPanel
+          datasource={mockDatasource()}
+          setState={setState}
+          state={{ ...state, x: { ...state.x, showGridlines } }}
+        />
+      );
+
+      (testSubj(component, 'lnsXY_xShowGridlines').onChange as Function)();
+
+      expect(setState).toHaveBeenCalledTimes(1);
+      return setState.mock.calls[0][0];
+    };
+
+    expect(toggleXGridlines(true)).toMatchObject({
+      x: { showGridlines: false },
+    });
+    expect(toggleXGridlines(false)).toMatchObject({
+      x: { showGridlines: true },
+    });
+  });
+
+  test('allows editing the y axis title', () => {
+    const testSetTitle = (title: string) => {
+      const setState = jest.fn();
+      const component = mount(
+        <XYConfigPanel datasource={mockDatasource()} setState={setState} state={testState()} />
+      );
+
+      (testSubj(component, 'lnsXY_yTitle').onChange as Function)({ target: { value: title } });
+
+      expect(setState).toHaveBeenCalledTimes(1);
+      return setState.mock.calls[0][0];
+    };
+
+    expect(testSetTitle('Hoi')).toMatchObject({
+      y: { title: 'Hoi' },
+    });
+    expect(testSetTitle('There!')).toMatchObject({
+      y: { title: 'There!' },
+    });
+  });
+
+  test('the y dimension panel accepts numeric operations', () => {
+    const datasource = {
+      ...mockDatasource(),
+      renderDimensionPanel: jest.fn(),
+    };
+    const state = testState();
+    const component = mount(
+      <XYConfigPanel
+        datasource={datasource}
+        setState={jest.fn()}
+        state={{ ...state, y: { ...state.y, accessors: ['a', 'b', 'c'] } }}
+      />
+    );
+
+    const panel = testSubj(component, 'lnsXY_yDimensionPanel_a');
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
nit: Instead of any you can use
```
    const nativeProps = (panel as NativeRendererProps<DatasourceDimensionPanelProps>).nativeProps;
```

You don't need the eslint-rule, so it's not much longer character-wise ;)